parsing/location.mli

```diff
@@ -26,7 +26,27 @@ type t = Warnings.loc = {
   loc_start: Lexing.position;
   loc_end: Lexing.position;
   loc_ghost: bool;
-}
+  }
+(** [t] represents a segment of source code between two positions.
+
+    For positions coming from a parser, loc_ghost is usually false, which
+    indicates that the information attached to the location could in principle
+    be parsed from the location. This is used by tools like merlin that want to
+    relate source code with parsetrees or later asts.
+
+    Example: in `let f x = x`, we have:
+    - a structure item at location "let f x = x"
+    - a pattern "f" at location "f"
+    - an expression "fun x -> x" at location "x = x"
+    - a pattern "x" at location "x"
+    - an expression "x" at location "x"
+    In this case, every node has loc_ghost=false, except the node "fun x -> x",
+    since [Parser.expression (Lexing.from_string "x = x")] would fail to parse.
+    By contrast, in `let f = fun x -> x`, every node has loc_ghost=false.
```

Comment:
A possibly simpler example would be `a.(b)` which is parsed to subtrees
- `a`
- `Array.get` at ghost location "a.(x)" 
- `b`

Reply:
I could add this example if you want to, but I think the more tricky example is also useful.